editor/editor_inspector.cpp

```diff
@@ -1702,7 +1702,7 @@ void EditorInspector::update_tree() {
 					ep->update_property();
 					ep->update_reload_status();
 
-					if (current_selected && ep->property == current_selected) {
+					if (current_selected && ep->property == current_selected && has_focus()) {
```

Comment:
This will be always false because the inspector never gets focus. What you want to do is right in intention though.

I would do the following:
1) Check if there is a focused control, you can do this by doing get_focus_owner() from any control.
2) if there is, check that any of the parents is an inspector, a while loop will work like:
```
bool restore_focus=false;
Control *focused = get_focus_owner();
if (focused) {
  Control * parent = focused->get_parent()
  while(parent) {
     EditorInspector *inspector = Object::cast_to<EditorInspector>(parent);
     if (inspector) { 
          restore_focus=inspector == this; //may be owned by another inspector
          break; //exit after the first inspector is found, since there may be nested ones
       }
      parent = parent->get_parent();
   }
}

if (current_selected && ep->property == current_selected && restore_focus) {
```